File: solutions/php/sum-of-multiples/1/SumOfMultiples.php
<?php
declare(strict_types=1);

function sumOfMultiples(int $number, array $multiples): int
{
    $sum = 0;
    for ($i = 1; $i < $number; $i++) {
        foreach ($multiples as $multiple) {
            if ($multiple === 0) {
                continue;
            }
            if ($i % $multiple === 0) {
                $sum += $i;
                break;
            }
        }
    }
    return $sum;
}

File: solutions/lua/sum-of-multiples/1/sum-of-multiples.lua
return function(numbers)
    numbers = numbers or { 3, 5 }

    return {
        to = function(limit)
            local sum = 0
            for i = 1, limit - 1 do
                for _, number in ipairs(numbers) do
                    if i % number == 0 then
                        sum = sum + i
                        break
                    end
                end
            end
            return sum
        end
    }
end
